src/NUnitFramework/framework/Constraints/Comparers/EquatablesComparer.cs

```diff
@@ -59,6 +64,18 @@ internal EquatablesComparer(NUnitEqualityComparer equalityComparer)
             return null;
         }
 
+        private static bool DoesUseStructuralEquality(Type type)
+        {
+            if (!type.IsValueType)
+                return false;
```

Comment:
Why would a reference type not use structural equality? For example, C# 9 records are emitted as ordinary classes, but they implement structural equality as indeed any class can if it wishes.